docs/hello_nextflow/02_hello_world.md

```diff
@@ -333,115 +410,135 @@ nextflow run hello-world.nf -resume
 The console output should look similar.
 
 ```console title="Output"
-N E X T F L O W  ~  version 23.10.1
-Launching `hello-world.nf` [tiny_elion] DSL2 - revision: 7ad1cd6bfe
-executor >  local (1)
-[8b/1f9ded] process > sayHello [100%] 1 of 1 ✔, cached: 1 ✔
+ N E X T F L O W   ~  version 24.02.0-edge
+
+ ┃ Launching `hello-world.nf` [thirsty_gautier] DSL2 - revision: 6654bc1327
+
+[10/15498d] sayHello [100%] 1 of 1, cached: 1 ✔
 ```
 
-Notice the additional `cached`. Nextflow has cached the process and re-used the result. It will also not replace the output file at `results/output.txt`.
+Notice the additional `cached:` bit in the process status line, which means that Nextflow has recognized that it has already done this work and simply re-used the result from the last run.
+
+!!! note
+
+    When your re-run a pipeline with `resume`, Nextflow does not overwrite any files written to a publishDir directory by any process call that was previously run successfully.
 
 ### Takeaway
 
-You know how to to relaunch a pipeline without repeating steps that were already executed in an identical way.
+You know how to to relaunch a pipeline without repeating steps that were already run in an identical way.
 
 ### What's next?
 
 Learn how to add in variable inputs.
 
 ---
 
-## 5. Add in variable inputs
+## 5. Add in variable inputs using a channel
 
-So far, we've been emitting a greeting hardcoded into the process command. Now we're going to add some flexibility by introducing channels.
+So far, we've been emitting a greeting hardcoded into the process command.
+Now we're going to add some flexibility by using an input variable, so that we can easily change the greeting.
+This is going to require us to use a **channel**; more on that in a minute.
```

Comment:
```suggestion
Now we're going to add some flexibility by using an input variable, so that we can easily change the greeting.

This requires us to make a series of inter-related changes:

1. Tell the process about expected variable inputs using the `input:` block
2. Edit the process to use the input
3. Create a **channel** to pass input to the process (more on that in a minute)
4. Add the channel as input to the process call
```

I like the `tell them what you're going to tell them before you tell them' school of thought, maybe something like the above would be good? I was momentarily confused by the 'four changes' thing lower down.